sklearn/multioutput.py

```diff
@@ -719,6 +736,15 @@ def fit(self, X, Y, **fit_params):
         self : object
             Returns a fitted instance.
         """
+        # TODO(1.8): Remove when base_estimator is removed
+        if self.base_estimator != "deprecated":
+
+            warning_text = (
+                "`base_estimator` was renamed to `estimator` in 1.6 "
+                "and will be removed in 1.8."
+            )
+            warnings.warn(warning_text, FutureWarning)
+            self.estimator = self.base_estimator
```

Comment:
According to our API guides, we should never change a given input constructor argument.

Here we should assign the estimator to a local variable and use that instead.

We should also check whether both `estimator` and `base_estimator` are provided, and error in that case.

Reply:
The behaviour should also be tested

Reply:
> We should also check whether both `estimator` and `base_estimator` are provided, and error in that case.

what kind of error should be raised

Reply:
> According to our API guides, we should never change a given input constructor argument.

Do you mean, to never change the value of `self.estimator` or `self.base_estimator`

Reply:
See https://github.com/scikit-learn/scikit-learn/pull/23819 for an example of a similar PR to this one.

Reply:
@adrinjalali has this conversation been resolved? I added a new internal attribute `self._estimator` and `self._validate_paramters()` that raises `ValueError` when both `base_estimator` and `estimator` are passed or when the `self._estimator` is None